frontend/packages/kubevirt-plugin/src/components/vms/vm-console.tsx

```diff
@@ -63,7 +68,31 @@ const VMConsoles: React.FC<VMConsolesProps> = ({
   serial,
   rdp,
   LoadingComponent,
+  type,
+  showOpenInNewWindow = true,
 }) => {
+  const [consoleType, setConsoleType] = React.useState(type);
```

Comment:
do we need the state's and useEffect here?

The type cannot be changed right? So shouldn't it be enough to compute this only on first render?